references/obj_detection/train_pytorch.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+
+os.environ['USE_TORCH'] = '1'
+
+import time
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from doctr.datasets import DocArtefacts
+from model import faster_model
+from utilities import *
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def fit_one_epoch(model_, images, targets):
+    x = images.to(device)
+    height, width = x.shape[0], x.shape[1]
+    for i in targets["boxes"]:
+        i[0] = round(i[0] * width)
+        i[2] = round(i[2] * width)
+        i[1] = round(i[1] * height)
+        i[2] = round(i[3] * height)
+    target_ = [{i: torch.tensor(j, dtype=torch.float32).to(device)} for i, j in targets.items()]
+    loss_dict = model_(x, target_)
+    loss = sum([l for k, l in loss_dict.items()])
+    return loss
+
+
+def evaluate(images, targets, val_metric, model_):
+    x, y = images.to(device), targets
+    targets = [torch.stack(j, dim=1)[0].to(device) for i, j in y.items()]
+    pq_metric, seg_quality, recall, precision = val_metric(targets=targets, model=model_, x=x)
+    return pq_metric, seg_quality, recall, precision
+
+
+def train_faster(optimizer, scheduler, train_dataloader, val_dataloader, mymodel: nn.Module, root_dir: str,
+                 label_path: str, num_epochs: int,
+                 val_metric):
+    """Training script
+    Args:
+        root_dir: location of image folder
+        label_path: location of label folder
+        num_epochs: number of epochs in training
+    Returns:
+        torch checkpoint
+    """
+    epch_pq = 0.0
+    epch_precision = 0.0
+    epch_recall = 0.0
+    epch_seg = 0.0
+    epch_train_loss = None
+    epch_loss = None
```

Comment:
Those need to be initialized inside `evaluate`

Reply:
Thanks, changes has been made :)